### PR TITLE
Use correct header_mappings_dir for subspecs

### PR DESCRIPTION
**Pull request by drcapulet**

### CODE REVIEW

File: lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb [hunk 1]
```diff
@@ -90,7 +90,8 @@ def install!
                 unless skip_info_plist?(native_target)
                   create_info_plist_file(target.info_plist_path, native_target, target.version, target.platform, :additional_entries => target.info_plist_entries)
                 end
-                create_build_phase_to_symlink_header_folders(native_target)
+                target_file_accessor = target.file_accessors.detect { |fa| fa.spec.name == target.name }
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: so this is meant to be always from the root spec?

Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: what happens if for a PodTarget only subspecs are enabled and not the root spec? Is this build phase completely missing?

Comment (drcapulet) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: After thinking about it more, if any header_mapping_dirs are present, we'll need to have copy phases for all headers, right?

Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: yes I think thats right.

File: CHANGELOG.md [hunk 2]
```diff
@@ -65,6 +65,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Use correct `header_mappings_dir` for subspecs
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit add two empty spaces at the end of this line for formatting.

File: lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb [hunk 3]
```diff
@@ -855,15 +855,21 @@ def project_file_references_array(files, file_type)
...
-          def header_mappings_dir
+          def any_header_mapping_dirs?
+            return @any_header_mapping_dirs if defined?(@any_header_mapping_dirs)
+            @any_header_mapping_dirs = target.file_accessors.any? { |fa| fa.spec_consumer.header_mappings_dir }
+          end
+
+          def header_mappings_dir(file_accessor)
+            @header_mappings_dirs ||= {}
+            return @header_mappings_dirs[file_accessor] if @header_mappings_dirs.key?(file_accessor)
             return @header_mappings_dir if defined?(@header_mappings_dir)
```
Comment (segiddins) [inline lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: this line is now obsolete?

Comment (drcapulet) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: Whoops, fixed!

**Pull request merged**